runtime/doc/help.txt

```diff
@@ -185,6 +185,7 @@ Other ~
 |channel.txt|		Nvim asynchronous IO
 |dev_style.txt|		Nvim style guide
 |job_control.txt|	Spawn and control multiple processes
+|luarefvim.txt|		Lua reference manual
```

Comment:
I'd use just `luaref.txt` (or `luareference.txt`, for disambiguation) -- the `vim` part is slightly misleading since the documentation is _not_ Vim-related (only the format is).